internal/gitrepo/gitrepo.go

```diff
@@ -224,10 +224,13 @@ func PrintStatus(ctx context.Context, repo *Repo) error {
 // Returns the commits in an API rooted at the given path,
 // stopping looking at the given commit (which is not included in the results).
 // The returned commits are ordered such that the most recent commit is first.
-func GetApiCommits(ctx context.Context, repo *Repo, path string, commit string) ([]object.Commit, error) {
+func GetApiCommits(repo *Repo, path string, commit string, retrieveAfterTimestamp *time.Time) ([]object.Commit, error) {
```

Comment:
I suspect we should refactor this - let's work out what querying we want to support, then design the functions we want.

(I think a function of "does commit X affect path Y" would be useful - and then a method to find all commits matching a given predicate, with another predicate as a stopping point.)